src/ngraph/frontend/onnx_import/op/topk.cpp

```diff
@@ -20,25 +20,38 @@
 #include "ngraph/node.hpp"
 #include "ngraph/op/get_output_element.hpp"
 #include "ngraph/op/topk.hpp"
+#include "ngraph/shape.hpp"
 #include "ngraph/type/element_type.hpp"
 #include "topk.hpp"
 #include "utils/common.hpp"
+#include "utils/reshape.hpp"
 
+/// \return Parse node attribute value for axis and adjust for negative value if needed.
 static std::int64_t get_axis(const ngraph::onnx_import::Node& node)
 {
-    // Parse node attribute value for axis (adjust for negative value if needed).
     std::int64_t axis{node.get_attribute_value<std::int64_t>("axis", -1)};
 
     auto data = node.get_ng_inputs().at(0);
     auto data_rank = data->get_shape().size();
     return ngraph::onnx_import::common::validate_axis(node, axis, data_rank);
 }
 
-static ngraph::NodeVector get_outputs(const std::shared_ptr<ngraph::Node>& top_k)
+/// \return Return the second input to the TopK node reshaped to a scalar.
+static std::shared_ptr<ngraph::Node> get_k(const ngraph::onnx_import::Node& node)
 {
-    std::shared_ptr<ngraph::Node> indices =
-        std::make_shared<ngraph::op::GetOutputElement>(top_k, 0);
-    std::shared_ptr<ngraph::Node> values = std::make_shared<ngraph::op::GetOutputElement>(top_k, 1);
+    auto k_node = node.get_ng_inputs().at(1);
+    NGRAPH_CHECK(shape_size(k_node->get_shape()) == 1,
+                 "ONNX TopK operator: 'K' parameter must contain a single positive value.",
```

Comment:
The error message mentions the sign of a value but there's no check for a sign here.

Reply:
The value is unknown at the time of graph import. We're only checking the shape, but I think the informative description of the expected value (from ONNX spec) may be helpful. I can remove the word `single` if you insist.